internal/handlers/common/aggregations/stages/group.go

```diff
@@ -165,73 +171,102 @@ func (g *group) Process(ctx context.Context, iter types.DocumentsIterator, close
 
 // groupDocuments groups documents by group expression.
 func (g *group) groupDocuments(ctx context.Context, in []*types.Document) ([]groupedDocuments, error) {
-	groupKey, ok := g.groupExpression.(string)
-	if !ok {
-		// non-string key aggregates values of all `in` documents into one aggregated document.
-		return []groupedDocuments{{
-			groupID:   g.groupExpression,
-			documents: in,
-		}}, nil
-	}
+	switch groupKey := g.groupExpression.(type) {
+	case *types.Document:
+		if !operators.IsOperator(groupKey) {
+			break
+		}
 
-	expression, err := aggregations.NewExpression(groupKey)
-	if err != nil {
-		var exprErr *aggregations.ExpressionError
-		if !errors.As(err, &exprErr) {
-			return nil, lazyerrors.Error(err)
+		op, err := operators.NewOperator(groupKey)
+		if err != nil {
+			return nil, err
```

Comment:
Could you add a test for error case please? See how it's done for `$project` https://github.com/FerretDB/FerretDB/blob/4d534d602e82d7726f327c2915d73cc253195553/integration/aggregate_documents_test.go#L393-L428